File: python/2015/day02a/day02a.py
#!/usr/bin/env python3

import sys

def usage():
    progname = sys.argv[0]
    print(f'usage: {progname} <input file>')
    sys.exit(1)

def process(filename):
    totalArea = 0
    with open(filename, 'r') as infile:
        for line in infile:
            (l, w, h) = tuple(map(int, line.split('x')))
            area1 = l * w
            area2 = l * h
            area3 = w * h
            surfaceArea = (2 * area1) + (2 * area2) + (2 * area3)
            minArea = min(area1, area2, area3)
            totalArea += surfaceArea + minArea
    return totalArea

def main():
    if len(sys.argv) < 2:
        usage()
    filename = sys.argv[1]
    result = process(filename)
    print(f'result = {result}')

if __name__ == '__main__':
    main()

File: perl/2015/day02a/day02a.pl
#!/usr/bin/env perl

use strict;
use warnings;

use List::Util qw(min);

sub usage {
    my $progname = $0;
    print "usage: $progname <input file>\n";
    exit 1;
}

sub process {
    my ($filename) = @_;
    my $totalArea = 0;

    open ( my $infile, '<', $filename ) or die "cannot open file: $filename: $!";
    while ( my $line = <$infile> ) {
	    my ($l, $w, $h) = map { $_ + 0 } split(/x/, $line);
	    my $area1 = $l * $w;
	    my $area2 = $l * $h;
	    my $area3 = $w * $h;
	    my $surfaceArea = (2 * $area1) + (2 * $area2) + (2 * $area3);
	    my $minArea = min($area1, $area2, $area3);
	    $totalArea += $surfaceArea + $minArea;
    }
    close($infile);

    return $totalArea;
}

sub main {
    if ( @ARGV < 1 ) {
        usage();
    }
    my $filename = $ARGV[0];
    my $result = process($filename);
    print "result = $result\n";
}

main();
